--- a/src/aggregator.rs
+++ b/src/aggregator.rs
@@ -84,7 +84,6 @@
             .connect_timeout(Duration::from_secs(config.request_timeout as u64)) // Add timeout to request to avoid DDOSing the server
             .use_rustls_tls()
             .tls_built_in_root_certs(config.operating_system_tls_certificates)
-            .https_only(true)
             .gzip(true)
             .brotli(true)
             .http2_adaptive_window(config.adaptive_window);
